addons/web/static/src/core/advanced_select/advanced_select.js

```diff
@@ -0,0 +1,123 @@
+/** @odoo-module **/
+
+import { Component, useState, useRef } from "@odoo/owl";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { _lt } from "@web/core/l10n/translation";
+import { useDebounced } from "@web/core/utils/timing";
+import { scrollTo } from "@web/core/utils/scrolling";
+
+export class AdvancedSelect extends Component {
+    setup() {
+        this.nextGroupId = 0;
+        this.nextOptionId = 0;
+
+        this.state = useState({ groups: [] });
+        this.inputRef = useRef("inputRef");
+        this.debouncedOnInput = useDebounced(this.onInput, 250);
+    }
+
+    onOpened(scrollContainer) {
+        if (this.inputRef.el) {
+            this.inputRef.el.focus();
+        }
+
+        if (scrollContainer) {
+            const element = scrollContainer.querySelector(".o_select_active");
+            if (element) {
+                scrollTo(element, { scrollable: scrollContainer });
+            }
+        }
+    }
+
+    isOptionSelected(option) {
+        return this.props.value === option.value;
+    }
+
+    getDropdownItemClass(option) {
+        return ("o_select_item px-0 " +
+            (this.isOptionSelected(option)
+                ? "o_select_active bg-light text-primary fw-bolder fst-italic"
+                : "")
+        );
```

Comment:
nitpicking: I don't like the trailing space for all the not selected options.
Of course there are other alternatives than mine:
```suggestion
const classes = ["o_select_item", "px-0"];
if (this.isOptionSelected(option)) {
    classes.push(...[
        "o_select_active",
        "bg-light",
        "text-primary",
        "fw-bolder",
        "fst-italic",
    ]);
}
return classes.join(" ");
```